aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -1271,11 +1272,27 @@ Tensor _embedding_bag_backward(const Tensor &grad, const Tensor &indices_,
 
   Tensor offset2bag_;
   if (indices.numel() != 0 && offset2bag.numel() == 0) {
-    offset2bag_ = at::zeros(
+    // For Composite Compliance, if `offsets` is CCT then
+    // we want `offset2bag_` to be CCT as well because
+    // `make_offset2bag` calls `index_add_` on `offset2bag_`
+    // with `offsets` as one of the arguments.
+    if (isTensorSubclassLike(offsets)) {
+      offset2bag_ = offsets.new_zeros(
+       {indices.size(0) + 1}, offsets.options()); // offset2bag = [0 0 0 0 0]
+    } else {
+      offset2bag_ = at::zeros(
        {indices.size(0) + 1}, offsets.options()); // offset2bag = [0 0 0 0 0]
+    }
```

Comment:
Could we replace these lines with:
```
offset2bag_ = offsets.new_zeros({indices.size(0) + 1});
```
This should do the same as both branches, right?

Reply:
Makes sense. Updated :) Thanks!